### PR TITLE
lds: cidr equality only checks significant bits

### PR DESCRIPTION
**Pull request by stevenctl**
Partially solves https://github.com/istio/istio/issues/29197. The specific case laid out there has CIDRs that have a different prefix, but the prefix length only includes equal parts.

This will make sure we accurately detect conflicting filter chains.

### CODE REVIEW

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -648,14 +648,31 @@ func CidrRangeSliceEqual(a, b []*core.CidrRange) bool {
...
-		if a[i].GetAddressPrefix() != b[i].GetAddressPrefix() || a[i].GetPrefixLen().GetValue() != b[i].GetPrefixLen().GetValue() {
+		netA, err := toIPNet(a[i])
+		if err != nil {
+			return false
+		}
+		netB, err := toIPNet(b[i])
+		if err != nil {
+			return false
+		}
+		fmt.Println(netA, netB)
```
Comment (howardjohn) [inline pilot/pkg/networking/util/util.go:659]: remove

**Review by howardjohn — APPROVED** [review]

**Comment by stevenctl** [comment]
I'll add a release note when the larger bug is fixed.

**Pull request merged**